tiledb/sm/tile/tile_metadata_generator.h

```diff
@@ -158,45 +184,20 @@ class TileMetadataGenerator {
   static bool has_sum_metadata(
       const Datatype type, const bool var_size, const uint64_t cell_val_num);
 
-  /**
-   * Returns the min and max of a fixed data tile.
-   *
-   * @param type_data Type data struct.
-   * @param tile Tile to process.
-   * @param cell_size Cell size.
-   *
-   * @return minimum, maximum.
-   */
-  template <class T>
-  static const tuple<void*, void*> min_max(
-      const Tile& tile, const uint64_t cell_size);
-
-  /**
-   * Returns the min and max of a fixed data tile with nullable values.
-   *
-   * @param type_data Type data struct.
-   * @param tile Tile to process.
-   * @param tile_validity Validity tile.
-   * @param cell_size Cell size.
-   *
-   * @return minimum, maximum, null count.
-   */
-  template <class T>
-  static const tuple<void*, void*, uint64_t> min_max_nullable(
-      const Tile& tile, const Tile& tile_validity, const uint64_t cell_size);
-
   /* ********************************* */
   /*     CONSTRUCTORS & DESTRUCTORS    */
```

Comment:
It might be worth adding a second constructor that takes an array schema and string, and uses that to set the other parameters. This is how it is being initialized in both spots in the writer base and in the dense tiler, so it would reduce a non-trivial chunk of repeated code.